exporter/elasticsearchexporter/internal/serializer/otelserializer/profile.go

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelserializer // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer"
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/elasticsearchexporter/internal/serializer/otelserializer/serializeprofiles"
+)
+
+const (
+	allEventsIndex   = "profiling-events-all"
+	stackTraceIndex  = "profiling-stacktraces"
+	stackFrameIndex  = "profiling-stackframes"
+	executablesIndex = "profiling-executables"
+)
+
+// SerializeProfile serializes a profile and calls the `pushData` callback for each generated document.
+func SerializeProfile(resource pcommon.Resource, scope pcommon.InstrumentationScope, profile pprofile.Profile, pushData func(*bytes.Buffer, string, string) error) error {
+	data, err := serializeprofiles.Transform(resource, scope, profile)
+	if err != nil {
+		return err
+	}
+
+	for _, payload := range data {
+		event := payload.StackTraceEvent
+
+		if event.StackTraceID != "" {
+			c, err := toJSON(event)
+			if err != nil {
+				return err
+			}
+			err = pushData(c, "", allEventsIndex)
```

Comment:
The downsampling and writing to `profiling-events-5pow*` is missing.

Reply:
This would be a blocker for beta/stable/production use. But isn't for development, and we have confirmed the exporter works properly without it.
So we're going without this for now, and will add it in a follow-up PR.